asapdiscovery-alchemy/asapdiscovery/alchemy/schema/network.py

```diff
@@ -74,6 +74,28 @@ def get_planning_function(self) -> Callable:
         )
 
 
+class CustomNetworkPlanner(_NetworkPlannerMethod):
+    """Plan a user-set custom network"""
+
+    type: Literal["CustomNetworkPlanner"] = "CustomNetworkPlanner"
+
+    custom_network_file: str = Field(
+        None,
+        description="The path to a CSV file containing edges specified by the user. Each line contains an edge, and ligand names (corresponding to the names in the input SDF) are separated by commas.",
+    )
+
+    def get_planning_function(self) -> Callable:
+        return openfe.ligand_network_planning.generate_network_from_names
+
+    def read_custom_network_csv(csv_path):
+        import csv
```

Comment:
Should use pandas rather than raw `csv` 

Also need to check 

- [ ] shape of resultant df, error or odd shape
- [ ]  type of data, error on non string
- [ ] Error if empty